### ISSUE TITLE
Inviting a user to a room via email does not create an invite in Synapse

### ISSUE THREAD

**Issue by anoadragon453** [matrix-org/synapse#7335]
### Description

Trying to invite a new user to a room via their email address allows them to sign up to the homeserver, but they don't actually get an invite to the room once they're signed in.

### Steps to reproduce

- `@alice:example.com` invites `bob@bobshouse.com` to a direct chat.
- bob registers, setting their email to `bob@bobshouse.com` and username to `bob`
- no room invite occurs inside the server (and this isn't even over federated synapses)

cc @michaelkaye 

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**: vector.modular.im

<!--
 What version of Synapse is running?

You can find the Synapse version with this command:

$ curl http://localhost:8008/_synapse/admin/v1/server_version

(You may need to replace `localhost:8008` if Synapse is not configured to
listen on that port.)
-->
- **Version**: v1.12.3

**Comment by babolivier** [comment]
I tried that with develop just this morning and I got the invite as soon as the 3PID of the newly registered user got bound, so not sure what's happening here. Though I did the registration and the bind separately, so maybe the call to /register doesn't result in a call to /bind on the IS (which might be a consequence of the privacy sprint?). AFAIK 3PID invites are only delivered upon a successful call to /bind on the IS so if that doesn't happen the user won't get pending invites.
